### PR TITLE
Add SHA checksum

### PR DESCRIPTION
**Pull request by J0WI**
Fixes https://github.com/docker-library/official-images/pull/7288#issuecomment-625555257

### CODE REVIEW

File: Dockerfile-alpine.template [hunk 1]
```diff
@@ -81,19 +81,21 @@ RUN set -ex; \
     \
     export GNUPGHOME="$(mktemp -d)"; \
     export GPGKEY="3D06A59ECE730EB71B511C17CE752F178259BD92"; \
-    curl -fsSL -o phpMyAdmin.tar.xz $URL; \
-    curl -fsSL -o phpMyAdmin.tar.xz.asc $URL.asc; \
+    curl -fsSL -o phpMyAdmin-$VERSION-all-languages.tar.xz $URL; \
+    curl -fsSL -o phpMyAdmin-$VERSION-all-languages.tar.xz.asc $URL.asc; \
+    curl -fsSL -o phpMyAdmin-$VERSION-all-languages.tar.xz.sha256 $URL.sha256; \
```
Comment (tianon) [inline Dockerfile-alpine.template:86]: Downloading the checksum during _build_ doesn't provide any value (as you've pointed out).  This should be downloaded during update and embedded into the templated file.

**Comment by emanuelb** [comment]
I meant adding checksum like wordpress does using ENV variable:
https://github.com/docker-library/wordpress/blob/master/php7.4/apache/Dockerfile#L98
which use SHA1 that's problematic, I opened issue about it https://github.com/docker-library/wordpress/issues/494
otherwise it's still vulnerable to downgrade attack as the gpg verification done on the archive file and not on the checksum .sha256 file, it's easy to test the attack by using higher version variable, such as:
`ENV VERSION 5.0.2`
and older URL value (to simulate a downgrade attack) that use files from older version: 5.01
`ENV URL https://files.phpmyadmin.net/phpMyAdmin/5.0.1/phpMyAdmin-5.0.1-all-languages.tar.xz`
if the build is successful it's vulnerable.

File: Dockerfile-alpine.template (hunk 1 above)
Comment (J0WI) [reply Dockerfile-alpine.template]: That actually makes more sense.

**Comment by emanuelb** [comment]
improvements to above change:
1. the * char is not needed at echo command:
`echo "$SHA256 *phpMyAdmin.tar.xz" | sha256sum -c -; \`
2. strip non [a-f0-9] chars from sha256 variable in update.sh and verify the result length is 64.

**Comment by emanuelb** [comment]
I stand corrected :) was taking into assumption that both methods are equivalent as shown in:
https://www.unix.com/man-page/centos/1/sha256sum
> Note: There is no difference between binary and text mode option on GNU system.

but as you say, it doesn't mean it won't have difference in other implementation, so its better to be explicit about the mode used, I marked the above suggestion as deleted.

**Comment by williamdes** [comment]
So I can merge this one ?

**Comment by emanuelb** [comment]
Yes, regarding the 2 improvement it can be done for example by adding:
` | tr -cd 'a-f0-9' | cut -c 1-64`

see diff:

```diff
-sha256="$(curl -fsSL "https://files.phpmyadmin.net/phpMyAdmin/$latest/phpMyAdmin-$latest-all-languages.tar.xz.sha256" | cut -f1 -d ' ')"
+sha256="$(curl -fsSL "https://files.phpmyadmin.net/phpMyAdmin/$latest/phpMyAdmin-$latest-all-languages.tar.xz.sha256" | cut -f1 -d ' ' | tr -cd 'a-f0-9' | cut -c 1-64)"
```

**Comment by williamdes** [comment]
Merged !
Thank you for the work and reviews :+1: 
I added 85a642d7b38bb824223ef3c4cfa6fc080d10acf3 to apply the suggestion of @emanuelb

**Pull request merged**